### PR TITLE
Remove Dashing, Eloquent, Foxy, L500 and SR300 support

### PR DESCRIPTION
**Pull request by SamerKhshiboun**
Remove EOL models
Remove EOL distros
Remove unused files/code (specially related to fisheye and pose sensors)

### CODE REVIEW

**Comment by Nir-Az** [comment]
This is draft right?
Please update when it's ready for review.

**Comment by Arun-Prasad-V** [comment]
Hi @SamerKhshiboun ,

Since POSE stream is getting removed, does it make sense to remove the support for POSE sensor as well? 
Reference line:
https://github.com/IntelRealSense/realsense-ros/blob/19bc683074f8d757fec8925c15aea7317d401728/realsense2_camera/src/rs_node_setup.cpp#L156

**Comment by SamerKhshiboun** [comment]
> sensor.is<rs2::pose_sensor>()

Fixed. Also deleted PoseProfile usage in other places

**Comment by Arun-Prasad-V** [comment]
> > sensor.isrs2::pose_sensor()
> 
> Fixed. Also deleted PoseProfile usage in other places

@SamerKhshiboun, It's still present in profile_manager.cpp
https://github.com/IntelRealSense/realsense-ros/blob/844210d353ab8665bd245511edd9de19dbc383bf/realsense2_camera/src/profile_manager.cpp#L596

**Comment by SamerKhshiboun** [comment]
> > > sensor.isrs2::pose_sensor()
> > 
> > 
> > Fixed. Also deleted PoseProfile usage in other places
> 
> @SamerKhshiboun, It's still present in profile_manager.cpp
> 
> https://github.com/IntelRealSense/realsense-ros/blob/844210d353ab8665bd245511edd9de19dbc383bf/realsense2_camera/src/profile_manager.cpp#L596

Fixed, somehow it was missed in the git add command

File: realsense2_camera/scripts/rs2_listener.py [hunk 1]
```diff
@@ -19,18 +19,12 @@
...
-if (os.getenv('ROS_DISTRO') != "dashing"):
-    import tf2_ros
-if (os.getenv('ROS_DISTRO') == "humble"):
-    from sensor_msgs.msg import PointCloud2 as msg_PointCloud2
-    from sensor_msgs_py import point_cloud2 as pc2
-# from sensor_msgs.msg import PointCloud2 as msg_PointCloud2
-# import sensor_msgs.point_cloud2 as pc2
+import tf2_ros
+from sensor_msgs.msg import PointCloud2 as msg_PointCloud2
+from sensor_msgs_py import point_cloud2 as pc2
```
Comment (Nir-Az) [inline realsense2_camera/scripts/rs2_listener.py:27]: This used to be `humble` only, is it relevant to `iron` and `rolling` too?

**Comment by Nir-Az** [comment]
@SamerKhshiboun please solve conflicts `This branch has conflicts that must be resolved`

**Comment by Arun-Prasad-V** [comment]
@SamerKhshiboun , Shall we remove the support of 'fisheye' sensor as well?
https://github.com/IntelRealSense/realsense-ros/blob/8656492a7d8ac3931cd7d1c4349bf07d73c89d78/realsense2_camera/src/rs_node_setup.cpp#L146
https://github.com/IntelRealSense/realsense-ros/blob/8656492a7d8ac3931cd7d1c4349bf07d73c89d78/realsense2_camera/src/rs_node_setup.cpp#L332

Also, in rs_launch.py
https://github.com/IntelRealSense/realsense-ros/blob/8656492a7d8ac3931cd7d1c4349bf07d73c89d78/realsense2_camera/launch/rs_launch.py#L47

https://github.com/IntelRealSense/realsense-ros/blob/8656492a7d8ac3931cd7d1c4349bf07d73c89d78/realsense2_description/urdf/_r430.urdf.xacro#L35

File: realsense2_camera/scripts/rs2_listener.py (hunk 1 above)
Comment (SamerKhshiboun) [reply realsense2_camera/scripts/rs2_listener.py:27]: did not understand the question.. can you elaborate more ? are you talking about pc2 or tf2_ros ?

**Comment by SamerKhshiboun** [comment]
> @SamerKhshiboun , Shall we remove the support of 'fisheye' sensor as well?
> 
> https://github.com/IntelRealSense/realsense-ros/blob/8656492a7d8ac3931cd7d1c4349bf07d73c89d78/realsense2_camera/src/rs_node_setup.cpp#L146
> 
> 
> https://github.com/IntelRealSense/realsense-ros/blob/8656492a7d8ac3931cd7d1c4349bf07d73c89d78/realsense2_camera/src/rs_node_setup.cpp#L332
> 
> Also, in rs_launch.py
> 
> https://github.com/IntelRealSense/realsense-ros/blob/8656492a7d8ac3931cd7d1c4349bf07d73c89d78/realsense2_camera/launch/rs_launch.py#L47
> 
> https://github.com/IntelRealSense/realsense-ros/blob/8656492a7d8ac3931cd7d1c4349bf07d73c89d78/realsense2_description/urdf/_r430.urdf.xacro#L35

Fixed

**Review by Arun-Prasad-V — APPROVED** [review]

File: realsense2_camera/scripts/rs2_listener.py (hunk 1 above)
Comment (Nir-Az) [reply realsense2_camera/scripts/rs2_listener.py:27]: NVM I understand now :)

**Review by Nir-Az — APPROVED** [review]
LGTM

**Comment by christian-rauch** [comment]
Is there a good reason to break the compatibility with `foxy`? Couldn't this be handled with #ifdefs like the other distributions? Could we alternatively at least introduce branches for distributions that are incompatible with the development branch? E.g. create a `foxy` branch for that last commit that still compiles on ROS `foxy`?

**Pull request merged**